StyleCop.Analyzers/StyleCop.Analyzers.Test.CSharp7/MaintainabilityRules/SA1414CSharp7UnitTests.cs

```diff
@@ -117,5 +118,67 @@ public TestClass({typeExpression} p1)
 
             await VerifyCSharpDiagnosticAsync(testCode, expected, CancellationToken.None).ConfigureAwait(false);
         }
+
+        [Fact]
+        public Task ValidateTuplesFromInterfaceAsync()
+        {
+            const string testCode = @"
+using System.Collections.Generic;
+
+namespace Test {
+    class StringTupleComparer : IEqualityComparer<(string, string)>
+    {
+	    public bool Equals((string, string) x, (string, string) y) => throw null;
+
+	    public int GetHashCode((string, string) obj) => throw null;
+    }
+}";
+            return VerifyCSharpDiagnosticAsync(testCode, DiagnosticResult.EmptyDiagnosticResults, default);
+        }
+
+        [Fact]
+        public Task ValidateTuplesFromExplicitInterfaceImplementationAsync()
+        {
+            const string testCode = @"
+using System.Collections.Generic;
+
+namespace Test {
+    class StringTupleComparer : IEqualityComparer<(string, string)>
+    {
+	    bool IEqualityComparer<(string, string)>.Equals((string, string) x, (string, string) y) => throw null;
+
+	    int IEqualityComparer<(string, string)>.GetHashCode((string, string) obj) => throw null;
+    }
+}";
+            return VerifyCSharpDiagnosticAsync(testCode, DiagnosticResult.EmptyDiagnosticResults, default);
+        }
+
+        [Fact]
+        public Task ValidateTuplesFromBaseClassAsync()
+        {
+            const string testCode = @"
+namespace Test {
+    class A : B
+	{
+		public override void Run((string, string) x)
+		{
+		}
+
+		public override void Run((int, int) y)
+		{
+		}
+	}
+
+	abstract class B
+	{
+		public abstract void Run(([|string|], [|string|]) x);
+
+		public virtual void Run(([|int|], [|int|]) y)
+		{
+		}
+	}
+}";
+            return VerifyCSharpDiagnosticAsync(testCode, Array.Empty<DiagnosticResult>(), default);
```

Comment:
I can't see any real downside to `Array.Empty<DiagnosticResult>()`, but `DiagnosticResult.EmptyDiagnosticResults` is used everywhere else.